src/Symfony/Component/ErrorHandler/ErrorRenderer/HtmlErrorRenderer.php

```diff
@@ -52,10 +52,10 @@ public function __construct(bool|callable $debug = false, string $charset = null
     {
         $this->debug = \is_bool($debug) ? $debug : $debug(...);
         $this->charset = $charset ?: (\ini_get('default_charset') ?: 'UTF-8');
-        $fileLinkFormat ??= $_ENV['SYMFONY_IDE'] ?? $_SERVER['SYMFONY_IDE'] ?? null;
-        $this->fileLinkFormat = \is_string($fileLinkFormat)
-            ? (ErrorRendererInterface::IDE_LINK_FORMATS[$fileLinkFormat] ?? $fileLinkFormat ?: false)
-            : ($fileLinkFormat ?: \ini_get('xdebug.file_link_format') ?: get_cfg_var('xdebug.file_link_format') ?: false);
+        $this->fileLinkFormat = \is_string($fileLinkFormat) ? new FileLinkFormatter($fileLinkFormat) : ($fileLinkFormat ?: false);
+        if (false === $this->fileLinkFormat) {
+            $this->fileLinkFormat = new FileLinkFormatter();
+        }
```

Comment:
The type of `$fileLinkFormat` is `string|FileLinkFormatter|null`. And the `FileLinkFormatter` constructor accepts `string|array|null`. The assignment can be one-lined.

```suggestion
        $this->fileLinkFormat = $fileLinkFormat instanceof FileLinkFormatter ? $fileLinkFormat : new FileLinkFormatter($fileLinkFormat);
```

---

But, note the class `FileLinkFormatter` can be missing as the `symfony/error-handler` package doesn't require `symfony/http-kernel`.

Reply:
Thanks for you review! I must say these one liners gives me headaches 🙂  Sure they are concise but [takes minutes to process](https://github.com/zakirullin/cognitive-load) for my (modest/slow) developer brain 🤯

Reply:
Sorry for your brain. That's actually less logic and operations.

The type of `$fileLinkFormat;` can be modified to `private FileLinkFormatter $fileLinkFormat;`

Reply:
Done!